### PR TITLE
feat!: typescript migration, adaptation to new peak-shape-generator

### PR DESCRIPTION
**Pull request by jobo322**
it still needs the new release of ml-peak-shape-generator

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,114 +2,33 @@
 
 [![NPM version][npm-image]][npm-url]
 [![build status][ci-image]][ci-url]
+[![Test coverage][codecov-image]][codecov-url]
 [![npm download][download-image]][download-url]
 
-Generates a spectrum from discrete peaks. The shape of the peaks can be customized.
-
-The shape is generated using [ml-peak-shape-generator](https://github.com/mljs/peak-shape-generator) and you may use all the corresponding [options](https://mljs.github.io/peak-shape-generator/#getshape) of getShape.
+.
```
Comment (targos) [inline README.md:8]: revert

File: README.md [hunk 2]
```diff
@@ -2,114 +2,33 @@
...
 
 ## Installation
 
 `$ npm i spectrum-generator`
 
 ## Usage
 
-### generateSpectrum
```
Comment (targos) [inline README.md:17]: revert

File: package.json [hunk 3]
```diff
@@ -1,61 +1,65 @@
 {
-  "name": "spectrum-generator",
```
Comment (targos) [inline package.json:2]: don't remove the name field

File: spectrum-generator.d.ts [hunk 4]
```diff
@@ -1,181 +0,0 @@
-export interface SpectrumGeneratorOptions {
```
Comment (targos) [inline spectrum-generator.d.ts:1]: can you at least copy the jsdoc from this file into the new typescript files?

Comment (jobo322) [reply spectrum-generator.d.ts:1]: Is it ok if I put the documentation similar than peak-shape-generator?

**Review by lpatiny — COMMENTED** [review]
All the types that could be defined and exported in ml-shape-generator should be in this other project.

This includes ShapeKind, Shape2DKind, ShapeOptions (for 1D and 2D), XYNumber, etc.

BTW please rename XYNumber to DataXY to stay consistant with the ml-spectra-processing package.

**Comment by jobo322** [comment]
> All the types that could be defined and exported in ml-shape-generator should be in this other project.
> 
> This includes ShapeKind, Shape2DKind, ShapeOptions (for 1D and 2D), XYNumber, etc.
> 
> BTW please rename XYNumber to DataXY to stay consistant with the ml-spectra-processing package.



> All the types that could be defined and exported in ml-shape-generator should be in this other project.
> 
> This includes ShapeKind, Shape2DKind, ShapeOptions (for 1D and 2D), XYNumber, etc.
> 
> BTW please rename XYNumber to DataXY to stay consistant with the ml-spectra-processing package.

I see that DataXY is an object with x and y properties as array unlike XYNumber that is numbers

**Pull request merged**